pkg/build/api/v1/types.go

```diff
@@ -163,6 +163,9 @@ type BuildSource struct {
 	// repository.
 	ContextDir string `json:"contextDir,omitempty" description:"specifies sub-directory where the source code for the application exists, allows for sources to be built from a directory other than the root of a repository"`
 
+	// DockerfileName is the name of the Dockerfile that will be used to build the Docker image.
+	DockerfileName string `json:"dockerfileName,omitempty" description:"specifies the name of the Dockerfile to use for building the Docker image"`
```

Comment:
I haven't looked carefully at the Dockerfile input feature lately, but why would you need to provide a dockerfile name when you're going through that flow?  To ensure you overwrite any existing Dockerfile provided in the input source?

if so i would say that makes an argument for providing the dockerfile name in the input source.